Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp

```diff
@@ -821,6 +827,44 @@ const char * ObjectNode::GetObjExtension() const
 	return m_ObjExtensionOverride.Get();
 }
 
+// ContainsWarningsMSCL
+//------------------------------------------------------------------------------
+bool ObjectNode::ContainsWarningsMSCL(const char * data, uint32_t dataSize) const
+{
+	ASSERT( IsMSVC() );
+	bool containsWarnings = false;
+	static const char warningString[] = "warning";
+
+	if( ( data == nullptr ) || ( dataSize == 0 ) )
+	{
+		return containsWarnings;
+	}
+
```

Comment:
We could take advantage of the fact that the buffer returned by Process is null terminated so we could replace most of this logic with:
```C++
return ( strstr(data, "\nwarning") != nullptr );
```
I need to do some tests to see if this string is localized though.